lib/elastomer/client/scroller.rb

```diff
@@ -128,19 +135,14 @@ class Scroller
       #
       def initialize( client, query, opts = {} )
         @client = client
-        @query  = query
 
-        @index       = opts.fetch(:index, nil)
-        @type        = opts.fetch(:type, nil)
-        @scroll      = opts.fetch(:scroll, '5m')
-        @size        = opts.fetch(:size, 50)
-        @search_type = opts.fetch(:search_type, nil)
+        @opts = DEFAULT_OPTS.merge({ :body => query }).merge(opts)
 
         @scroll_id = nil
         @offset = 0
       end
 
-      attr_reader :client, :query, :index, :type, :scroll, :size, :search_type, :scroll_id
+      attr_reader :client, :query, :scroll_id
```

Comment:
Did you check to see if there are any uses of these reader methods? We may want to continue supporting them to avoid breakage.